src/edit_session/bracket_match.js

```diff
@@ -219,5 +219,190 @@ function BracketMatch() {
         
         return null;
     };
+
+    /**
+     * Returns [[Range]]'s for matching tags and tag names, if there are any
+     * @param {Position} pos
+     * @returns {{closeTag: Range, closeTagName: Range, openTag: Range, openTagName: Range} | undefined}
+     */
+    this.getMatchingTags = function (pos) {
+        var iterator = new TokenIterator(this, pos.row, pos.column);
+        var token = iterator.getCurrentToken();
+        var found = false;
+        var backward = false;
+        if (token && token.type.indexOf('tag-name') === -1) {
+            do {
+                if (backward) token = iterator.stepBackward(); else token = iterator.stepForward();
+                if (token) {
+                    if (token.value === "/>") backward = true;
+                    if (token.type.indexOf('tag-name') !== -1) {
+                        found = true;
+                    }
+                }
+            } while (token && !found);
+        }
+        if (!token) return;
+
+        var tag = token.value;
+        var currentTag = token.value;
+        var depth = 0;
+        var prevToken = iterator.stepBackward();
+
+        if (prevToken.value === '<') {
+            //find closing tag
+            var openTagStart = new Range(iterator.getCurrentTokenRow(), iterator.getCurrentTokenColumn(),
+                iterator.getCurrentTokenRow(), iterator.getCurrentTokenColumn() + 1
+            );
+            token = iterator.stepForward();
+            var openTagName = new Range(iterator.getCurrentTokenRow(), iterator.getCurrentTokenColumn(),
+                iterator.getCurrentTokenRow(), iterator.getCurrentTokenColumn() + token.value.length
+            );
+            iterator.stepBackward();
+            var foundOpenTagEnd = false;
+            do {
```

Comment:
this loop and function seems a bit too complex to understand. Could you refactor it out into separate methods / functions which are easier to read?

Reply:
I used logic from `$highlightTags` method and evolved it to support both closing and opening ranges. Will try to break it into some logical blocks. Thank you!

Reply:
I hope that would work